aggregator_core/src/datastore.rs

```diff
@@ -6025,6 +6089,9 @@ pub enum Error {
     /// An invalid parameter was provided.
     #[error("invalid parameter: {0}")]
     InvalidParameter(&'static str),
+    /// A time type (duration, interval, timestamp) was not aligned to the task precision.
+    #[error("time unaligned to task precision")]
+    TimeUnaligned,
```

Comment:
It's unfortunate to have to discard underlying errors everywhere. If you don't want to commit to doing `TimeUnaligned(janus_messages::Error)`, another option is `TimeUnaligned(Box<dyn Error>)`, which would still preserve error chains and let us print them even if we can't destructure them. `TimeUnaligned(String)` also works (then you'd do something like `Error::TimeUnaligned(format!("{:?}", underlying_error)`), but `Box<dyn Error>` is nicer and either way, it's a variable length heap allocation.

Reply:
Fixed in 3369f41d